### PR TITLE
feat: Add support for Nim

### LINKED ISSUE #159: Add Nim on starship
This programming language is close to v1.0 release (currently v1.0 RC2) and it is used by some companies, for example Status is implementing an Ethereum 2.0 client.
Its syntax is python-ish and has a package manager called Nimble.
Website: https://nim-lang.org/

### PR DESCRIPTION
**Pull request by jmcduffie32**
#### Description

Add Nim language version module

#### Motivation and Context

Closes #159 

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ x] I have updated the documentation accordingly.
- [ x] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/nim.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+use std::process::Command;
+
+use super::{Context, Module};
+
+/// Creates a module with the current Nim version
+///
+/// Will display the Nim version if any of the following criteria are met:
+///     - Current directory contains a `.nim` file
+///     - Current directory contains a `.nimble` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
```
Comment (chipbuster) [inline src/modules/nim.rs]: This looks fine to me, but I'd like to have someone else have a look at these files, as language modules haven't really been the area I've worked on in Starship.

File: tests/testsuite/nim.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+use ansi_term::Color;
```
Comment (chipbuster) [inline tests/testsuite/nim.rs]: 👍

File: docs/config/README.md [hunk 3]
```diff
@@ -467,6 +467,30 @@ The module will be shown if any of the following conditions are met:
 symbol = "⚙️ "
 ```
 
+## Nim
```
Comment (iamsauravsharma) [inline docs/config/README.md]: I think Nim should be in ascending order. All modules are listed in are in ascending order currently

File: src/modules/mod.rs [hunk 4]
```diff
@@ -8,6 +8,7 @@ mod git_status;
 mod golang;
 mod jobs;
 mod line_break;
+mod nim;
```
Comment (iamsauravsharma) [inline src/modules/mod.rs]: You have not updated const ALL_MODULES_LIST present in src/module.rs according to this update

File: src/print.rs [hunk 5]
```diff
@@ -22,6 +22,7 @@ const DEFAULT_PROMPT_ORDER: &[&str] = &[
     "rust",
     "python",
     "golang",
+    "nim",
```
Comment (iamsauravsharma) [inline src/print.rs]: You haven't updated this in config README to include this update

**Comment by matchai** [comment]
Thank you for taking the time to really try to get this working.
Getting the installation working in CI really is a pain, and I just want you to know that this isn't going unnoticed.

When we get around to tackling #289, we will no longer need to install external runtimes to get our tests running, so we can get back to this one once that has been addressed. 😊

**Comment by jmcduffie32** [comment]
Thanks @matchai; I'll keep an an eye on #289

**Comment by matchai** [comment]
#289 has been merged and is now being used when testing new modules, though the PR will also need to be updated to support the new module structure.

Closing the PR as stale for now. Feel free to open a new one 😊

**Pull request closed without merging**